core/src/test/java/org/opensearch/sql/expression/json/JsonFunctionsTest.java

```diff
@@ -61,4 +76,239 @@ private ExprValue execute(ExprValue jsonString) {
+    objectMap.put("bar2", new ExprDoubleValue(12.34));
+    objectMap.put("baz", ExprNullValue.of());
+    objectMap.put(
+        "obj", ExprTupleValue.fromExprValueMap(Map.of("internal", new ExprStringValue("value"))));
+    objectMap.put(
+        "arr",
+        new ExprCollectionValue(
+            List.of(new ExprStringValue("string"), ExprBooleanValue.of(true), ExprNullValue.of())));
+    ExprValue expectedTupleExpr = ExprTupleValue.fromExprValueMap(objectMap);
+
+    // exercise
+    exp = DSL.json_function(DSL.literal(objectJson));
+
+    // Verify
+    var value = exp.valueOf();
+    assertTrue(value instanceof ExprTupleValue);
+    assertEquals(expectedTupleExpr, value);
+  }
+
+  @Test
+  void json_returnsJsonArray() {
+    FunctionExpression exp;
+
+    // Setup
+    final String arrayJson = "[\"foo\", \"fuzz\", true, \"bar\", 1234, 12.34, null]";
+    ExprValue expectedArrayExpr =
+        new ExprCollectionValue(
+            List.of(
+                new ExprStringValue("foo"),
+                new ExprStringValue("fuzz"),
+                LITERAL_TRUE,
+                new ExprStringValue("bar"),
+                new ExprIntegerValue(1234),
+                new ExprDoubleValue(12.34),
+                LITERAL_NULL));
+
+    // exercise
+    exp = DSL.json_function(DSL.literal(arrayJson));
+
+    // Verify
+    var value = exp.valueOf();
+    assertTrue(value instanceof ExprCollectionValue);
+    assertEquals(expectedArrayExpr, value);
+  }
+
+  @Test
+  void json_returnsScalar() {
+    assertEquals(
+        new ExprStringValue("foobar"), DSL.json_function(DSL.literal("\"foobar\"")).valueOf());
+
+    assertEquals(new ExprIntegerValue(1234), DSL.json_function(DSL.literal("1234")).valueOf());
+
+    assertEquals(LITERAL_TRUE, DSL.json_function(DSL.literal("true")).valueOf());
+
+    assertEquals(LITERAL_NULL, DSL.json_function(DSL.literal("null")).valueOf());
+
+    assertEquals(LITERAL_NULL, DSL.json_function(DSL.literal("")).valueOf());
+
+    assertEquals(
+        ExprTupleValue.fromExprValueMap(Map.of()), DSL.json_function(DSL.literal("{}")).valueOf());
+  }
+
+  @Test
+  void json_returnsSemanticCheckException() {
+    // invalid type
+    assertThrows(
+        SemanticCheckException.class, () -> DSL.castJson(DSL.literal("invalid")).valueOf());
+
+    // missing bracket
+    assertThrows(SemanticCheckException.class, () -> DSL.castJson(DSL.literal("{{[}}")).valueOf());
+
+    // missing quote
+    assertThrows(
+        SemanticCheckException.class, () -> DSL.castJson(DSL.literal("\"missing quote")).valueOf());
+  }
+
+  @Test
+  void json_extract_search() {
+    Expression jsonArray = DSL.literal(ExprValueUtils.stringValue("{\"a\":1}"));
+    ExprValue expectedExprValue = new ExprIntegerValue(1);
+    Expression pathExpr = DSL.literal(ExprValueUtils.stringValue("$.a"));
+    FunctionExpression expression = DSL.jsonExtract(jsonArray, pathExpr);
+    assertEquals(expectedExprValue, expression.valueOf());
```

Comment:
Nits.

Seems pretty inconsistent throughout this file, most of which you didn't write ... but might make sense to be consistent in how we create expression values (here _and_ and the other `json_extract_*` test methods)? Right now, for example, we are using _both_ `ExprValueUtils.stringValue()` and `new ExprStringValue()`. Would it make sense to just use one of these? To my eyes, just using `ExprValueUtils` is probably preferable because it would avoid us having to import every `Expr*Value` class (which is the main point of this utility class, I think).

Also, to make this more readable, might make sense to separate expected and actual cases, and name them consistently?

Lastly, is this a `jsonArray`? Perhaps `jsonExpr` could be more accurate? Or `jsonStringExpr`? 🤷 

```suggestion
    ExprValue expectedExprValue = ExprValueUtils.integerValue(1);
    
    Expression jsonStringExpr = DSL.literal(ExprValueUtils.stringValue("{\"a\":1}"));
    Expression pathStringExpr = DSL.literal(ExprValueUtils.stringValue("$.a"));
    ExprValue actualExprValue = DSL.jsonExtract(jsonStringExpr, pathStringExpr).valueOf();
    
    assertEquals(expectedExprValue, actualExprValue);
```

Reply:
One more thought: would it make sense to extract a private method like `test_json_extract` that you could use to simplify these tests?

```java
private static void `test_json_extract(String jsonString, String pathString, ExprValue expected) {
    Expression jsonStringExpr = DSL.literal(ExprValueUtils.stringValue(jsonString));
    Expression pathStringExpr = DSL.literal(ExprValueUtils.stringValue(pathString));
    ExprValue actualExprValue = DSL.jsonExtract(jsonStringExpr, pathStringExpr).valueOf();

   assertEquals(expectedExprValue, actualExprValue);
}
```

This could simplify these test methods a lot perhaps? 🤷

Reply:
added helper